services/controllerhost/consumer.go

```diff
@@ -568,6 +589,144 @@ func refreshCGExtents(context *Context,
 	return addExtentsToConsumerGroup(context, dstID, cgID, cgDesc.GetIsMultiZone(), newExtents, outputHosts, m3Scope), nil
 }
 
+func refreshCGExtentsKafka(context *Context,
```

Comment:
And instead of forking at the refreshCGExtents level, we can fork at selectNextExtentsToConsume level, which was the original way the code was written.